contracts/command-batcher/src/encoding/evm.rs

```diff
@@ -0,0 +1,467 @@
+            (vec![], vec![], vec![]),
+            |(mut addresses, mut weights, mut signatures), operator| {
+                addresses.push(Token::Address(ethereum_types::Address::from_slice(
+                    operator.address.as_slice(),
+                )));
+                weights.push(Token::Uint(ethereum_types::U256::from_big_endian(
+                    &operator.weight.to_be_bytes(),
+                )));
+
+                if let Some(signature) = &operator.signature {
+                    signatures.push(Token::Bytes(signature.into()));
+                }
+
+                (addresses, weights, signatures)
+            },
+        );
+
+        let threshold = Token::Uint(ethereum_types::U256::from_big_endian(
+            &multisig.quorum.to_be_bytes(),
+        ));
+
+        Ok(ethabi::encode(&[
+            Token::Array(addresses),
+            Token::Array(weights),
+            threshold,
+            Token::Array(signatures),
+        ])
+        .into())
+    }
+}
+
+#[cw_serde]
+pub struct Data {
+    pub destination_chain_id: Uint256,
+    pub commands: Vec<Command>,
+}
+
+impl Data {
+    pub fn encode(&self) -> HexBinary {
+        let destination_chain_id = Token::Uint(ethereum_types::U256::from_big_endian(
+            &self.destination_chain_id.to_be_bytes(),
+        ));
+
+        let (commands_ids, commands_types, commands_params) = self.commands.iter().fold(
+            (vec![], vec![], vec![]),
+            |(mut commands_ids, mut commands_types, mut commands_params), command| {
+                commands_ids.push(Token::FixedBytes(command.id.to_vec()));
+                commands_types.push(Token::String(command.ty.to_string()));
+                commands_params.push(Token::Bytes(command.params.to_vec()));
+                (commands_ids, commands_types, commands_params)
+            },
+        );
+
+        ethabi::encode(&[
+            destination_chain_id,
+            Token::Array(commands_ids),
+            Token::Array(commands_types),
+            Token::Array(commands_params),
+        ])
+        .into()
+    }
+}
+
+fn evm_address(pub_key: &[u8]) -> Result<HexBinary, ContractError> {
+    let pub_key =
+        PublicKey::from_sec1_bytes(pub_key).map_err(|e| ContractError::InvalidMessage {
+            reason: e.to_string(),
+        })?;
+    let pub_key = pub_key.to_encoded_point(false);
+
+    Ok(Keccak256::digest(&pub_key.as_bytes()[1..]).as_slice()[12..].into())
+}
+
+fn sorted_operators(signers: Vec<Signer>) -> Result<Vec<Operator>, ContractError> {
+    let mut operators = signers
+        .into_iter()
+        .map(|signer| {
+            Ok(Operator {
+                address: evm_address((&signer.pub_key).into())?,
+                weight: signer.weight,
+                signature: signer.signature,
+            })
+        })
+        .collect::<Result<Vec<Operator>, ContractError>>()?;
+    operators.sort_by(|a, b| a.address.cmp(&b.address));
```

Comment:
I'd suggest implementing `std::cmp::Ord` for `Operator` and then just use `.sort`.